BitBucket Repository Push (#311)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md

---
sidebar_position: 10
description: Keep track of latest commits in your catalog
---

> import CommitBlueprint from './resources/bitbucket/\_example_bitbucket_push_blueprint.mdx'

Reviewer: Let's rename to `MicroserviceBlueprint`

import CommitWebhookConfig from './resources/bitbucket/\_example_bitbucket_push_webhook_configuration.mdx'

# BitBucket

> In this example you are going to create a webhook integration between [BitBucket](https://bitbucket.org) and Port, which will ingest commit entities.

Reviewer: This isn't phrase well, please rephrase it in a way that will explain we will add to an existing microservices in port a data about the latest commit pushes to the default branch

Reviewer: In this example, we will be integrating BitBucket and Port to enhance an existing microservice in Port with data regarding the latest commit pushes to the default branch. This integration will involve setting up a webhook to receive notifications from BitBucket whenever a new commit is made to the default branch, allowing Port to ingest and process the commit entities accordingly.

Reviewer: something like that maybe

Author: thanks for this

## Prerequisites

> Create the following blueprint definition and webhook configuration:

Reviewer: ```suggestion
Create the following blueprint definition and webhook configuration in Port:
```

<details>

> <summary>Commit blueprint</summary>

Reviewer: Here as well

<CommitBlueprint/>

</details>

<details>

> <summary>Commit webhook configuration</summary>

Reviewer: Maybe Push webhook configuration instead

> <summary>Microservice blueprint</summary>

Reviewer: Add a note that explaining that the blueprint microservice can be extended, and that's just an example

<CommitWebhookConfig/>

</details>

## Create the BitBucket webhook

1. Go to your desired repository in BitBucket;
2. Select **Repository Settings**;
3. Select **Webhooks**;
4. Click on **Add webhook**;
5. Input the following details:
   1. `Title` - enter a descriptive title for this webhook;

>    2. `URL` - enter the value of the `url` key you received after creating the webhook configuration;

Reviewer: Maybe rephrase it in a way it will make sense that it comes from Port (the URL)

   3. Be sure to keep the "Active" checkbox checked under **Status**;
   4. Under **Triggers**, select "Push";
6. Click on **Save** at the bottom of the page.

:::tip

> In order to view the different payloads and events available in BitBucket webhooks, [look here](https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#Push)

Reviewer: Specify here that the structure of the data can be found here, and it is used by the configuration in the webhook configuration

:::

> Done! any change you make to the repository will trigger a webhook event that BitBucket will send to the webhook URL provided by Port. Port will parse the events according to the mapping and update the catalog entities accordingly.

Reviewer: ```suggestion
Done! any push to the repository's main branch will trigger a webhook event that will be sent to the webhook URL provided by Port. Port will parse the events according to the mapping and update the catalog entities accordingly.
```
